### DISCUSSION TITLE
Development log

### DISCUSSION THREAD

**Discussion by nicvagn** [HarshilPatel007/YACS#7] [Show and tell]
I have been working on the pgn creation implementation. 

I renamed the chessboard as DrawChessBoard() from chessboard to draw_chessboard or something.

I am working on organizing the chessboard objects and the move_managers into self contained classes.

If you want a suggestion on what to work on, and are looking for a project, the top bar and menus and that boring Gui stuff could use some work. 

That is not fun work, and if you have other ideas that is kk.

Salute

**Comment by nicvagn** [comment]
I have checked out the engine branch.

**Reply by nicvagn** [reply]
You got the highlighting issue working?

**Reply by HarshilPatel007** [reply]
@nicvagn , not yet.

I think we've to reconsider the code structure.
The problem is,
I'm calling the `make_engine_move` when user selects the `white` play as engine.
But, when application starts, we call `DrawChessBoard` and don't know why `ChessBoardHightlightManager` is not calling?

For `engine vs engine` case, it's not working because, we call `engine_move` on a mouse click. Basically, after user makes a move. So, that's the reason it's not working.
